src/test/consensus/Validations_test.cpp

```diff
@@ -707,21 +707,45 @@ class Validations_test : public beast::unit_test::suite
         TestHarness harness(h.oracle);
         Node a = harness.makeNode();
 
+        // simple cases
         Ledger ledgerA = h["a"];
         BEAST_EXPECT(ValStatus::current == harness.add(a.validate(ledgerA)));
-        BEAST_EXPECT(harness.vals().numTrustedForLedger(ledgerA.id()));
+        BEAST_EXPECT(harness.vals().numTrustedForLedger(ledgerA.id()) == 1);
+        harness.vals().expire();
+        BEAST_EXPECT(harness.vals().numTrustedForLedger(ledgerA.id()) == 1);
+        harness.clock().advance(harness.parms().validationSET_EXPIRES);
+        harness.vals().expire();
+        BEAST_EXPECT(harness.vals().numTrustedForLedger(ledgerA.id()) == 0);
 
-        // Keep the validation from expire
+        // use setSeqToKeep to keep the validation from expire
+        Ledger ledgerB = h["ab"];
+        BEAST_EXPECT(ValStatus::current == harness.add(a.validate(ledgerB)));
+        BEAST_EXPECT(harness.vals().numTrustedForLedger(ledgerB.id()) == 1);
+        harness.vals().setSeqToKeep(ledgerB.seq(), ++ledgerB.seq());
```

Comment:
This reads really weird for me. When I see `++`, I'm expecting to be incrementing something that will persist, like a variable. I get that it's legal, but it's confusing. I'd much rather see this written as `ledgerB.seq() + Ledger::Seq(1)`.

And while I'm thinking about it, I think most of these objects should be declared as `const` so it's clear that nothing is changing in them. `a`, `ledgerA`, `ledgerB`, `loops`, `ledgerC`.

Reply:
Thanks, changed.